### PR TITLE
exclude %encoded /

### PR DESCRIPTION
**Pull request by bourgeoa**
see issue #1666
excludes from %encode/%decode   `_exceptSlash () { return /(\/|%2f|%2F|%252F|%252f)/g }`

### CODE REVIEW

File: lib/resource-mapper.js [hunk 1]
```diff
@@ -149,6 +147,25 @@ class ResourceMapper {
     return { path, contentType: contentType || this._defaultContentType }
   }
 
+  // encode/decode path except slash/%encodedSlash/double%encodedSlash
+  _exceptSlash () { return /(\/|%2f|%2F|%252F|%252f)/g }
+
+  _encodePath (pathname) {
+    const pathArray = pathname.split(this._exceptSlash())
+    pathArray.forEach((el, i) => {
+      if (i % 2 === 0) pathArray[i] = encodeURIComponent(el)
```
Comment (RubenVerborgh) [inline lib/resource-mapper.js:156]: Might want to normalize path separators to lowercase or uppercase, such that they are mapped to the same filename. (see https://github.com/CommunitySolidServer/CommunitySolidServer/pull/1273)

File: lib/resource-mapper.js [hunk 2]
```diff
@@ -149,6 +147,25 @@ class ResourceMapper {
     return { path, contentType: contentType || this._defaultContentType }
   }
 
+  // encode/decode path except slash/%encodedSlash/double%encodedSlash
+  _exceptSlash () { return /(\/|%2f|%2F|%252F|%252f)/g }
```
Comment (RubenVerborgh) [inline lib/resource-mapper.js:151]: Good find about the `%25`. However, we found that we also need to guard against _sequences_ of `%25252525` in https://github.com/CommunitySolidServer/CommunitySolidServer/pull/1273.

File: lib/resource-mapper.js [hunk 3]
```diff
@@ -147,23 +147,29 @@ class ResourceMapper {
     return { path, contentType: contentType || this._defaultContentType }
   }
 
-  // encode/decode path except slash/%encodedSlash/double%encodedSlash
-  _exceptSlash () { return /(\/|%2f|%2F|%252F|%252f)/g }
+  // encode/decode path except slash (/) / %encodedSlash (%2F|%2f) /ntimes%encodedSlash (%2525...2F|%2525...2f)
```
Comment (TallTed) [inline lib/resource-mapper.js:150]:
```suggestion
  // encode/decode path except slash (/), %encodedSlash (%2F|%2f), and/or ntimes%encodedSlash (%2525...2F|%2525...2f)
```

File: test/unit/resource-mapper-test.js [hunk 4]
```diff
@@ -224,14 +224,14 @@ describe('ResourceMapper', () => {
         contentType: 'text/html'
       })
 
-    itMapsUrl(mapper, 'a URL of a new file with encoded characters',
+    itMapsUrl(mapper, 'a URL of a new file with encoded characters and encoded /',
```
Comment (TallTed) [inline test/unit/resource-mapper-test.js:227]:
```suggestion
    itMapsUrl(mapper, 'a URL of a new file with encoded characters and encoded slash ("/")',
```

**Comment by kjetilk** [comment]
I feel that I've been out of this code base for so long that I can't really provide a useful review at this level of detail. But LGTM.

**Comment by bourgeoa** [comment]
Already merged with https://github.com/nodeSolidServer/node-solid-server/pull/1695

**Pull request closed without merging**